### PR TITLE
Toggle details

### PR DESCRIPTION
**Pull request by ka1130**
This PR adds toggling details on "Show more" button click. This button appears below the show only if further data (summary) is available.

It also adds pagination to display maximum of 12 episodes per page.
The maximum limit of the container's width is set to avoid issues on large screens.

Both pagination and toggling details are done by hand, using React's features only. The same scope of issues was solved via Redux in [this PR](https://github.com/ka1130/TVloVer/pull/8)

Please advice which solution is better given the requirements. My guess is PR#8 and Redux would be better, however I don't know best practices well yet.

Screencast:
peek 2018-10-07 21-06

**Disclaimer**: _There's tons of comments in the code, it's just temporary and helps me understand the Redux flow. Hope they may stay until I get more comfortable with Redux. They will definitely be gone at further stages of work on the project._

### CODE REVIEW

File: src/components/App/EpisodesList/Show/Details/index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  return (
+    <div className={`${styles.detailsWrapper} ${styles.visible}`}>
+      <button
+        className={styles.toggleDetailsButton}
+        onClick={props.toggleDetails}
+      >
+        {!props.detailsShown ? 'Show details' : 'Hide details'}
+      </button>
+      <p className={`${props.detailsShown ? styles.visible : ''} ${styles.summary}`}>
+        {props.summary ? props.summary.replace('<p>','').replace('</p>','') : ''}
```
Comment (yakato) [inline src/components/App/EpisodesList/Show/Details/index.js]: I don't understand this `<p>` tag replacing. Won't it be easier to render conditionally Details button in the parent component? Or consider toggle details on click on the poster and add button for the Show link?

File: src/components/App/EpisodesList/Show/Details/index.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
```
Comment (yakato) [inline src/components/App/EpisodesList/Show/Details/index.js]: Are those details for Show or for Episode? Are you sure about nesting folders here?

File: src/components/App/Pagination/index.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    return pageNumbers.map(
+    number => <li 
+                key={number}
+                id={number}
+                onClick={this.props.handlePageChange}>
+                  {number}
+                </li>
+                )
+  }
+    render() {
```
Comment (yakato) [inline src/components/App/Pagination/index.js]: Indents here

File: src/components/App/Pagination/index.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React, { Component } from 'react';
+
+import styles from './Pagination.module.scss';
+
+class Pagination extends Component {
+  renderPageNumbers = () => {
+    const { episodes, episodesPerPage } = this.props;
+
+    let pageNumbers = [];
+    for (let i = 1; i <= Math.ceil(episodes.length / episodesPerPage); i++) {
```
Comment (yakato) [inline src/components/App/Pagination/index.js]: Maybe replace with `reduce`  function instead `for`?

File: src/components/App/Pagination/index.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      pageNumbers.push(i);
+    }
+
+    return pageNumbers.map(
+    number => <li 
+                key={number}
```
Comment (yakato) [inline src/components/App/Pagination/index.js]: There is lack of current page number style indicator to show where am I

File: src/components/App/index.js [hunk 6]
```diff
@@ -1,30 +1,50 @@
...
     if (error) {
       return <p>{error.message}</p>;
     }
 
     return (
       <div className={styles.appWrapper}>
         <Header/>
-        { loading ? <Spinner /> : <EpisodesList episodes={episodes} day={today}/> }
+        { loading ? <Spinner /> : <EpisodesList episodes={currentEpisodes} day={today}/> }
+        <Pagination episodes={episodes} episodesPerPage={episodesPerPage} handlePageChange={this.handlePageChange}/>
```
Comment (yakato) [inline src/components/App/index.js]: It should be render only when there are any episodes. Also please add some placeholder / help message for the user when there is no episodes for today

File: src/components/App/index.js [hunk 7]
```diff
@@ -1,30 +1,50 @@
...
 
+  handlePageChange = event => {
+    this.setState({ currentPage: Number(event.target.id)});
+  }
+
   render() {
     const { error, loading, episodes } = this.props;
+    const { currentPage, episodesPerPage } = this.state;
+
+    const indexOfLastEpisode = currentPage * episodesPerPage;
```
Comment (yakato) [inline src/components/App/index.js]: consider to hold everything for pagination in the store (current page number key) and then show only first 12, second 12 (...) filtered from the store data

Also you can consider keep current page in the url, what if I want to show someone third page and I'd love to have direct link? Pagination concept is not so easy, you need to think about cases and usage first. But 👍 for this basic implementation.

File: src/styles/base/_palette.scss [hunk 8]
```diff
@@ -0,0 +1,5 @@
+$white-smoke: #FBF5F3;
```
Comment (yakato) [inline src/styles/base/_palette.scss]: I like the color pallet in separate variables file 👍

File: src/components/App/EpisodesList/Show/Details/index.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+import styles from './Details.module.scss';
+
+const Details = props => {
+  if (!props.summary) return null;
+
+  return (
+    <div className={`${styles.detailsWrapper} ${styles.visible}`}>
```
[the remark this answers is not in the document]
Comment (ka1130) [reply src/components/App/EpisodesList/Show/Details/index.js]: Wow, that's a cool one! I'm introducing it, thanks! :D

**Comment by ka1130** [comment]
Closing because [PR#9](https://github.com/ka1130/TVloVer/pull/9) delivers the same features, refactored.

**Pull request closed without merging**